internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mappings.go

```diff
@@ -105,17 +98,18 @@ func dataSourceMongoDBAtlasFederatedSettingsOrganizationRoleMappingsRead(ctx con
 	return nil
 }
 
-func flattenFederatedSettingsOrganizationRoleMappings(federatedSettingsOrganizationRoleMapping *matlas.FederatedSettingsOrganizationRoleMappings) []map[string]any {
+func flattenFederatedSettingsOrganizationRoleMappings(federatedSettingsOrganizationRoleMapping *admin.PaginatedRoleMapping) []map[string]any {
```

Comment:
knit: I think the function can be simpliflied, e.g:

func flattenFederatedSettingsOrganizationRoleMappings(federatedSettingsOrganizationRoleMapping *admin.PaginatedRoleMapping) []map[string]any {
	var results []map[string]any
	mappings := federatedSettingsOrganizationRoleMapping.GetResults()
	for i := range mappings {
		mapping := &mappings[i]
		results = append(results, map[string]any{
			"external_group_name": mapping.GetExternalGroupName(),
			"id":                  mapping.GetId(),
			"role_assignments":    FlattenRoleAssignments(mapping.GetRoleAssignments()),
		})
	}
	return results
}

Reply:
Thanks. Applied in 41ee44e9ce9201e57951d72795c3949ddd31b83e ✅